### PR TITLE
H3 indices to points

### PR DESCRIPTION
**Pull request by shaardie**
This patch adds a new function to convert h3 indices back to longitude
and latitude.

It also adapts the aggregate function to use h3 as a column rather than
an index after grouping.

### CODE REVIEW

File: s5a/__init__.py [hunk 1]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
...
+    """Convert h3 index in dataframe into longitude and latitude.
+
+    :param: dataframe: pandas dataframe with a column of h3 indices
+    :type dataframe: pandas.core.frame.DataFrame
+    :param h3_column: column name of the column with the h3 indices,
+                      defaults to h3.
+    :type h3_column: str, optional
+    """
+    dataframe[['longitude', 'latitude']] = [
+        h3.h3_to_geo(h3grid) for h3grid in dataframe['h3']]
```
Comment (lkiesow) [inline s5a/__init__.py]: It's not a grid but a hexagon, isn't it?

```suggestion
        h3.h3_to_geo(h3hexagon) for h3hexagon in dataframe['h3']]
```

File: s5a/__init__.py [hunk 2]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
     return dataframe
 
 
+def h3_to_point(dataframe, h3_column='h3'):
+    """Convert h3 index in dataframe into longitude and latitude.
```
Comment (lkiesow) [inline s5a/__init__.py]:
```suggestion
    """Convert H3 index in pandas dataframe into longitude and latitude.
```

File: s5a/__init__.py [hunk 3]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
...
+
+    :param: dataframe: pandas dataframe with a column of h3 indices
```
Comment (lkiesow) [inline s5a/__init__.py]:
```suggestion
    :param: dataframe: pandas dataframe with a column of H3 indices
```

File: s5a/__init__.py [hunk 4]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
...
+    :type dataframe: pandas.core.frame.DataFrame
+    :param h3_column: column name of the column with the h3 indices,
```
Comment (lkiesow) [inline s5a/__init__.py]:
```suggestion
    :param h3_column: column name of the column with the H3 indices,
```

File: s5a/__init__.py [hunk 5]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
...
+                      defaults to h3.
```
Comment (lkiesow) [inline s5a/__init__.py:125]:
```suggestion
                      defaults to H3.
```

File: s5a/__init__.py [hunk 6]
```diff
@@ -97,6 +97,20 @@ def point_to_h3(dataframe, resolution=1):
...
+    :type h3_column: str, optional
+    """
+    dataframe[['longitude', 'latitude']] = [
+        h3.h3_to_geo(h3grid) for h3grid in dataframe['h3']]
+    return dataframe
+
+
 def aggregate_h3(dataframe, function='mean'):
     """Aggregate data values of the same h3 index in dataframe.
```
Comment (lkiesow) [inline s5a/__init__.py]:
```suggestion
    """Aggregate data values of the same H3 index in dataframe.
```

**Review by lkiesow — CHANGES_REQUESTED** [review]
Generally looks good. Just a few minor comments. I didn't test this though.

File: s5a/__init__.py (hunk 5 above)
Comment (shaardie) [reply s5a/__init__.py:125]: I actually defaults to `h3` :smile:

**Comment by shaardie** [comment]
@lkiesow: Please take a look at https://github.com/emissions-api/sentinel5algorithms/pull/27 first.

Then I can write tests for this before merging.

**Comment by shaardie** [comment]
I had to change this, since `dataframe[['longitude', 'latidute']]` breaks if those columns are not present.

Not quite sure if this is the fastest solution.

File: s5a/__init__.py (hunk 5 above)
Comment (lkiesow) [reply s5a/__init__.py:125]: Hah, makes sense. Then maybe something like:

```suggestion
                      defaults to ``h3``.
```

**Comment by lkiesow** [comment]
Let me know when I should take another look at this.

**Comment by shaardie** [comment]
@lkiesow You can take a look at this again. I know it is not in the scope of this Pull Request, but I also updated the existing tests and use `len()` instead of `.size`, since it is counting the rows and not all elements.

**Comment by tibroc** [comment]
@lkiesow this looks good to me, if it's okay with you I would merge it.

**Pull request merged**